.gitlab/includes/gcc12_cuda12_pipeline.yml

```diff
@@ -32,25 +32,26 @@ gcc12_cuda12_spack_image:
     - .variables_gcc12_cuda12_config
     - .dependencies_image_template_rosa
 
+# TODO: comment again the following steps when maintenance on clariden is over.
 # Test step currently commented as the cuda driver is too old on clariden:
 # https://github.com/pika-org/pika/issues/884
-#gcc12_cuda12_build:
-#  needs: [gcc12_cuda12_spack_image]
-#  extends:
-#    - .variables_gcc12_cuda12_config
-#    - .build_template_rosa
-#
-#.gcc12_cuda12_test_commmon:
-#  needs: [gcc12_cuda12_build]
-#  extends:
-#    - .variables_gcc12_cuda12_config
-#    - .test_common_gpu_clariden_cuda
-#    - .test_template
-#
-#gcc12_cuda12_test_release:
-#  extends: [.gcc12_cuda12_test_commmon]
-#  image: $PERSIST_IMAGE_NAME_RELEASE
-#
-#gcc12_cuda12_test_debug:
-#  extends: [.gcc12_cuda12_test_commmon]
-#  image: $PERSIST_IMAGE_NAME_DEBUG
+gcc12_cuda12_build:
+  needs: [gcc12_cuda12_spack_image]
+  extends:
+    - .variables_gcc12_cuda12_config
+    - .build_template_rosa
+
+.gcc12_cuda12_test_commmon:
+  needs: [gcc12_cuda12_build]
+  extends:
+    - .variables_gcc12_cuda12_config
+    - .test_common_gpu_daint_cuda
+    - .test_template
+
+gcc12_cuda12_test_release:
+  extends: [.gcc12_cuda12_test_commmon]
+  image: $PERSIST_IMAGE_NAME_RELEASE
+
+gcc12_cuda12_test_debug:
+  extends: [.gcc12_cuda12_test_commmon]
+  image: $PERSIST_IMAGE_NAME_DEBUG
```

Comment:
```suggestion
gcc12_cuda12_build:
  needs: [gcc12_cuda12_spack_image]
  extends:
    - .variables_gcc12_cuda12_config
    - .build_template_rosa

#.gcc12_cuda12_test_commmon:
#  needs: [gcc12_cuda12_build]
#  extends:
#    - .variables_gcc12_cuda12_config
#    - .test_common_gpu_daint_cuda
#    - .test_template
#
#gcc12_cuda12_test_release:
#  extends: [.gcc12_cuda12_test_commmon]
#  image: $PERSIST_IMAGE_NAME_RELEASE
#
#gcc12_cuda12_test_debug:
#  extends: [.gcc12_cuda12_test_commmon]
#  image: $PERSIST_IMAGE_NAME_DEBUG
```

The build step was actually wrongly commented out before. Perhaps you can move the comment about the test step being disabled down to the test step?